opengrok-indexer/src/main/java/org/opengrok/indexer/history/SubversionHistoryParser.java

```diff
@@ -87,19 +94,31 @@ private static class Handler extends DefaultHandler2 {
             sb = new StringBuilder();
         }
 
-        List<String> getRenamedFiles() {
-            return new ArrayList<>(renamedFiles);
+        Set<String> getRenamedFiles() {
+            return renamedFiles;
+        }
+
+        Map<String, String> getRenamedDirectories() {
+          return renamedToDirectoryRevisions;
         }
 
         @Override
         public void startElement(String uri, String localName, String qname, Attributes attr) {
-            isRenamed = false;
+            isRenamedFile = false;
+            isRenamedDir = false;
             if ("logentry".equals(qname)) {
                 entry = new HistoryEntry();
                 entry.setActive(true);
                 entry.setRevision(attr.getValue("revision"));
             } else if ("path".equals(qname)) {
-                isRenamed = attr.getIndex("copyfrom-path") != -1;
+                if (attr.getIndex(COPYFROM_PATH) != -1) {
+                  LOGGER.info("rename for:" + attr.getValue(COPYFROM_PATH) );
```

Comment:
could you reduce the level to e.g. `FINE` ? This would be too noisy in the logs, I think.

Reply:
actually, I will fix it after that fact.